parsons/redash/redash.py

```diff
@@ -0,0 +1,168 @@
+            raise Exception('Redash Query {} failed: {}'.format(query_id, job['error']))
+
+    def get_fresh_query_results(self, query_id, params=None):
+        """
+        Make a fresh query result and get back the CSV http response object back
+        with the CSV string in result.content
+
+        `Args:`
+            query_id: str or int
+                The query id of the query
+            params: dict
+                If there are parameters in the query (e.g. "{{datelimit}}" in the query),
+                then this is a dict that will pass the parameters in the POST.
+                We add the "p_" prefix for parameters, so if your query had ?p_datelimit=....
+                in the url, you should just set 'datelimit' in params here.
+        `Returns:`
+            Response Class
+        """
+        redash_params = ({'p_%s' % k: str(v).replace("'", "''") for k, v in params.items()}
+                         if params else {})
+
+        response = self.session.post(f'{self.base_url}/api/queries/{query_id}/refresh',
+                                     params=redash_params,
+                                     verify=self.verify)
+
+        if response.status_code != 200:
+            raise Exception(f'Refresh failed for query {query_id}. {response.text}')
+
+        job = response.json()['job']
+        result_id = self._poll_job(self.session, job, query_id)
+        if result_id:
+            response = self.session.get(
+                f'{self.base_url}/api/queries/{query_id}/results/{result_id}.csv',
+                verify=self.verify)
+            if response.status_code != 200:
+                raise Exception(f'Failed getting results for query {query_id}. {response.text}')
+        else:
+            raise Exception(f'Failed getting result {query_id}. {response.text}')
+        return response
+
+    def get_cached_query_results(self, query_id, query_api_key=None):
+        """
+        Get the results from a cached query result and get back the CSV http response object back
+        with the CSV string in result.content
+
+        `Args:`
+            query_id: str or int
+                The query id of the query
+            query_api_key: str
+                If you did not supply a user_api_key on the Redash object, then you can
+                supply a query_api_key to get cached results back anonymously.
+        `Returns:`
+            Response Class
+        """
+        params = {}
+        if not self.user_api_key and query_api_key:
+            params['api_key'] = query_api_key
+        response = self.session.get(f'{self.base_url}/api/queries/{query_id}/results.csv',
+                                    params=params,
+                                    verify=self.verify)
+        if response.status_code != 200:
+            raise Exception(f'Failed getting results for query {query_id}. {response.text}')
+        return response
+
+    @classmethod
+    def load_to_table(cls, refresh=True, **kwargs):
```

Comment:
I really like this pattern. We might have to look at incorporating this into some of our other Connectors.